NuRadioReco/modules/efieldGalacticNoiseAdder.py

```diff
@@ -0,0 +1,153 @@
+
+        for i_pixel in range(healpy.pixelfunc.nside2npix(self.__n_side)):
+            azimuth = local_coordinates[i_pixel].az.rad
+            zenith = np.pi / 2. - local_coordinates[i_pixel].alt.rad
+
+            if zenith > 90. * units.deg:
+                continue
+
+            temperature_interpolator = scipy.interpolate.interp1d(
+                self.__interpolation_frequencies, np.log10(self.__noise_temperatures[:, i_pixel]), kind='quadratic')
+            noise_temperature = np.power(10, temperature_interpolator(freqs[passband_filter]))
+
+            # calculate spectral radiance of radio signal using rayleigh-jeans law
+            spectral_radiance = (2. * (scipy.constants.Boltzmann * units.joule / units.kelvin)
+                                 * freqs[passband_filter] ** 2 * noise_temperature * self.solid_angle / c_vac ** 2)
+            spectral_radiance[np.isnan(spectral_radiance)] = 0
+
+            # calculate radiance per energy bin
+            spectral_radiance_per_bin = spectral_radiance * d_f
+
+            # calculate electric field per energy bin from the radiance per bin
+            efield_amplitude = np.sqrt(
+                spectral_radiance_per_bin / (c_vac * scipy.constants.epsilon_0 * (
+                        units.coulomb / units.V / units.m))) / d_f
+
+            # assign random phases to electric field
+            noise_spectrum = np.zeros((3, freqs.shape[0]), dtype=np.complex128)
+            phases = self.__random_generator.uniform(0, 2. * np.pi, len(spectral_radiance))
+
+            noise_spectrum[1][passband_filter] = np.exp(1j * phases) * efield_amplitude
+            noise_spectrum[2][passband_filter] = np.exp(1j * phases) * efield_amplitude
+
+            channel_noise_spec = np.zeros_like(noise_spectrum)
+
+            for field in station.get_electric_fields():
+                field_pos = field.get_position()
+                dt = geometryUtilities.get_time_delay_from_direction(
+                    zenith, azimuth, field_pos, n=n_air
+                )
+                delta_phases = -2 * np.pi * freqs[passband_filter] * dt
+
+                # add random polarizations and phase to electric field
+                polarizations = self.__random_generator.uniform(0, 2. * np.pi, len(spectral_radiance))
+
+                channel_noise_spec[1][passband_filter] = noise_spectrum[1][passband_filter] * np.exp(
+                    1j * delta_phases) * np.cos(polarizations)
+                channel_noise_spec[2][passband_filter] = noise_spectrum[2][passband_filter] * np.exp(
+                    1j * delta_phases) * np.sin(polarizations)
+
+                # add noise spectrum from this to field freq spectrum
+                field_spectra[field.get_unique_identifier()] += np.sum(channel_noise_spec, axis=0)
+
+        # Store updated fields spectra
+        for field in station.get_electric_fields():
+            field.set_frequency_spectrum(field_spectra[field.get_unique_identifier()], "same")
```

Comment:
Why du you store a single polarisation as the sum of all 3 polarisations instead of each polarisation independently?